streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimestampedStore.java

```diff
@@ -277,197 +275,361 @@ public void close() {
+        private Runnable closeCallback;
+        private byte[] noTimestampNext;
+        private byte[] withTimestampNext;
         private final Comparator<byte[]> comparator = Bytes.BYTES_LEXICO_COMPARATOR;
-
+        private final RocksIterator noTimestampIterator;
+        private final RocksIterator withTimestampIterator;
         private final String storeName;
-        private final RocksIterator iterWithTimestamp;
-        private final RocksIterator iterNoTimestamp;
         private final boolean forward;
-
+        private final boolean toInclusive;
+        private final byte[] rawLastKey;
         private volatile boolean open = true;
 
-        private byte[] nextWithTimestamp;
-        private byte[] nextNoTimestamp;
-        private KeyValue<Bytes, byte[]> next;
-        private Runnable closeCallback = null;
-
-        RocksDBDualCFIterator(final String storeName,
-                              final RocksIterator iterWithTimestamp,
-                              final RocksIterator iterNoTimestamp,
-                              final boolean forward) {
-            this.iterWithTimestamp = iterWithTimestamp;
-            this.iterNoTimestamp = iterNoTimestamp;
-            this.storeName = storeName;
+        /**
+         * Constructs a new {@code RocksDBDualCFRangeIterator}.
+         *
+         * <p>Initializes the RocksDB iterators for two column families (timestamped and non-timestamped) and sets up
+         * the range and direction for iteration.</p>
+         *
+         * @param from                  The starting key of the range. Can be {@code null} for an open range.
+         * @param to                    The ending key of the range. Can be {@code null} for an open range.
+         * @param noTimestampIterator   The iterator for the non-timestamped column family.
+         * @param withTimestampIterator The iterator for the timestamped column family.
+         * @param storeName             The name of the store associated with this iterator.
+         * @param forward               {@code true} for forward iteration; {@code false} for reverse iteration.
+         * @param toInclusive           Whether the upper boundary of the range is inclusive.
+         */
+        RocksDBDualCFRangeIterator(final Bytes from,
+                                   final Bytes to,
+                                   final RocksIterator noTimestampIterator,
+                                   final RocksIterator withTimestampIterator,
+                                   final String storeName,
+                                   final boolean forward,
+                                   final boolean toInclusive) {
             this.forward = forward;
+            this.noTimestampIterator = noTimestampIterator;
+            this.storeName = storeName;
+            this.toInclusive = toInclusive;
+            this.withTimestampIterator = withTimestampIterator;
+
+            this.rawLastKey = initializeIterators(from, to);
```

Comment:
generally it's better for constructors not to have side effects - otherwise it's possible that if something throws an exception during instantiation the `close` method won't be called.

this doesn't seem to be a regression (the old code did that as well for the RangeIterator) so consider this comment an optional suggestion

Reply:
Updated.